### PR TITLE
docs: add security section to x/mint README

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2070

### CODE REVIEW

File: x/mint/README.md [hunk 1]
```diff
@@ -102,6 +102,18 @@ oneYear := time.Duration(minttypes.NanosecondsPerYear)
...
 
+### Security
+
+Q: Can validators manipulate the amount of tokens minted due to inflation?
+
+A:
+
+This x/mint module calculates block provisions based on block timestamps so it is important to understand how block timestamps work in CometBFT. In CometBFT, block timestamps are monotonically increasing. A block's timestamp is the median of the `Vote.Time` fields from vote messages, where each vote's timestamp is weighted based on the voting power of the validator that cast it.
+
+Based on [BFT time](https://docs.cometbft.com/v0.34/spec/consensus/bft-time), a block's timestamp is manipulatable by malicious validators if they control > 1/3 of the total voting power. Consequently, if malicious validators control > 1/3 of the total voting power, they could manipulate a block's timestamp to some arbitrary value (e.g. one year in the future) effectively advancing the inflation schedule by a year.
```
Comment (SweeXordious) [inline x/mint/README.md:113]: > a block's timestamp is manipulatable by malicious validators if they control > 1/3 of the total voting power

[Question]
Why `>1/3`? Any validator can deviate the time for a certain portion except that it depends on their weight. If their weight is small, then the deviation would be small. If the weight is big, the deviation will be big.

**Review by evan-forbes — APPROVED** [review]
nice

File: x/mint/README.md (hunk 1 above)
Comment (SweeXordious) [reply x/mint/README.md:113]: Aah I see, thanks. But I still don't see how the 1/3 works:
> if malicious validators control > 1/3 of the total voting power, they could manipulate a block's timestamp to some arbitrary value (e.g. one year in the future)

I did a small test in here:
https://go.dev/play/p/bCku4cxDfu6

Comment (cmwaters) [reply x/mint/README.md:113]: The proposer of the next height chooses the signatures to add to the commit but they don't theoretically need to add all the signatures they see. They only need to have 2/3+ signatures. So imagine there's 100 voting power in total and 34 is malicious. I as the proposer of the next height include a last commit that contains the 34 malicious voting power and the 33 honest voting power (to get to 67). 34 > 33 so the weighted median is whatever the 34 malicious nodes set which could be 10 years in the future for example

**Review by cmwaters — APPROVED** [review]
Helpful explanation

File: x/mint/README.md (hunk 1 above)
Comment (SweeXordious) [reply x/mint/README.md:113]: Aaah I see, this was the missing point for me:
>  the next height chooses the signatures to add to the commit

Thanks

**Pull request merged**